spec/models/student_spec.rb

```diff
@@ -0,0 +1,326 @@
+  #     mock_repo = mock('Repository::AbstractRepository')
+  #     expect(mock_repo.stubs(:close)).to be true
+  #     expect(mock_repo.stubs(:remove_user)).to raise_error(Repository::UserNotFound)
+  #     expect(Group.any_instance.stubs(:repo)).to eq(mock_repo)
+  #
+  #     Student.hide_students(@student_id_list)
+  #
+  #     [{type: 'negative', grace_credits: '-10', expected: 0 },
+  #     {type: 'positive', grace_credits: '10', expected: 15 }].each do |item|
+  #       it "should not error when given #{item[:type]} grace credits" do
+  #         expect(Student.give_grace_credits(@student_id_list, item[:grace_credits]))
+  #
+  #         #You have to find the students to get the updated values
+  #         students = Student.find(@student_id_list)
+  #
+  #         expect(item[:expected]).eql?(students[0].grace_credits)
+  #         expect(item[:expected]).eql?(students[1].grace_credits)
+  #       end
+  #     end
+  #   end
+  # end
+
+  # context 'Hidden Students' do
+  #     @student1 = Student.new(:hidden)
+  #     @student2 = Student.new(:hidden)
+  #
+  #     @membership1 = StudentMembership.new({membership_status: StudentMembership::STATUSES[:inviter], user: @student1})
+  #     @grouping = @membership1.grouping
+  #     @membership2 = StudentMembership.new({grouping: @grouping, membership_status: StudentMembership::STATUSES[:accepted], user: @student2})
+  #
+  #     @student_id_list = [@student1.id, @student2.id]
+  #
+  #   it 'should unhide without error' do
+  #     #TODO test the repo with mocks
+  #     Student.unhide_students(@student_id_list)
+  #
+  #     students = Student.find(@student_id_list)
+  #     expect(students[0].hidden).to be false
+  #     expect(students[1].hidden).to be false
+  #   end
+  #
+  #   it 'should unhide without error when users already exists in repo' do
+  #     # Mocks to enter into the if
+  #     expect(Assignment.any_instance.stubs(:vcs_submit)).to be true
+  #     expect(Grouping.any_instance.stubs(:is_valid?)).to be true
+  #
+  #     # Mock the repository and raise Repository::UserNotFound
+  #     mock_repo = mock('Repository::AbstractRepository')
+  #     expect(mock_repo.stubs(:close)).to be true
+  #     expect(mock_repo.stubs(:add_user)).to raise_error(Repository::UserAlreadyExistent)
+  #     expect(Group.any_instance.stubs(:repo)).to eq(mock_repo)
+  #
+  #     Student.unhide_students(@student_id_list)
+  #   end
+  # end
+  #
+  # context 'A hidden Student' do
+  #     @student = Student.new(:hidden)
+  #
+  #     it 'should not become a member of a grouping' do
+  #       grouping = Grouping.new
+  #       @student.invite(grouping.id)
+  #
+  #       pending_memberships = @student.pending_memberships_for(grouping.assignment_id)
+  #
+  #       expect(pending_memberships).not_to be_nil
+  #       expect(pending_memberships.length).to eq(0)
+  #     end
+  #   end
+
+  context 'A Student' do
+    @student = Student.new
```

Comment:
Please use [FactoryBot](https://github.com/thoughtbot/factory_bot) as much as possible to create objects for testing.  Also, it is preferable to use [rspec's let and let!](https://relishapp.com/rspec/rspec-core/v/2-5/docs/helper-methods/let-and-let) methods instead of setting instance variables inside a context block.